### ISSUE TITLE
Confusion about packed vs. unpacked parameters for SVDQW4A4Linear

### ISSUE THREAD

**Issue by alexzms** [nunchaku-tech/nunchaku#770]
Hi Nunchaku developer team,

I'm working on quantizing a model using `SVDQW4A4Linear` and I'm seeing a large accuracy drop (high MSE) when using the Nunchaku engine, but not when using a manual PyTorch simulation. This suggests I am loading the parameters incorrectly.

The docstring for `SVDQW4A4Linear` states that `proj_down` and `proj_up` are "Packed".

### My Current (Failing) Approach

I am using `convert_to_nunchaku_w4x4y16_linear_weight` to get all the tensors. Then, I load them into the `SVDQW4A4Linear` module like this:

```python
@torch.no_grad()
def convert_linear_to_svdq(linear: nn.Linear, ...):
    
    # ... (SVD, compute scales, etc.) ...
    
    # 1. Get tensors from the official converter
    qweight, wscales_packed, bias_packed, smooth_packed, lora_packed, _ = convert_to_nunchaku_w4x4y16_linear_weight(
        weight=residual.to(torch_dtype),
        scale=wscales_4d.to(torch_dtype),
        bias=b_2d,
        smooth=sm_2d,
        lora=lora_tuple,
        ...
    )

    # 2. Build SVDQ layer
    svdq = SVDQW4A4Linear.from_linear(
        linear,
        rank=r_aligned,
        precision=precision,
        act_unsigned=act_unsigned,
    )

    # 3. Load parameters
    svdq.qweight.copy_(qweight)
    svdq.wscales.copy_(wscales_packed)
    
    # --- This is the part I am unsure about ---
    # Following the docstring, I'm loading the _packed versions
    if lora_packed is not None:
        svdq.proj_down.copy_(lora_packed[0])
        svdq.proj_up.copy_(lora_packed[1])
    
    svdq.bias.copy_(bias_packed)
    svdq.smooth_factor.copy_(smooth_packed)
    
    return svdq
```

When I run inference using this module (`qmodel(x)`), the loss is very high (MSE ≈ 1.95).

### My (Working) Manual Simulation

Separately, I have a manual PyTorch forward function that simulates the W4A4 math. This function uses the **unpacked** LoRA matrices (`lora_down`, `lora_up`), the **unpacked** bias (`linear.bias`), and the **unpacked** smooth factor.

This manual simulation works perfectly and gives a very low loss (MSE ≈ 0.0085).

### My Question

This discrepancy leads to my question: For the `SVDQW4A4Linear` CUDA kernels (`svdq_quantize...`, `svdq_gemm...`) to function correctly, should the following parameters be loaded with their **packed** or **unpacked** versions?

1.  `svdq.proj_down`
2.  `svdq.proj_up`
3.  `svdq.bias`
4.  `svdq.smooth_factor`

My successful manual test suggests they all need to be **unpacked**, which seems to contradict the docstring for `proj_down`/`proj_up`.

Could you please clarify the correct way to load these parameters into the `SVDQW4A4Linear` module?

@lmxyy @sxtyzhangzk  Thank you!

**Comment by alexzms** [comment]
@lmxyy @sxtyzhangzk Thank you!

**Comment by lillionfish** [comment]
Hi, I’m encountering the same issue as you. Did you manage to fix it, or find any workaround?
